test/0.8.9/accounting.handleOracleReport.test.ts

```diff
@@ -1,652 +1,409 @@
-  //     const accountingOracleSigner = await impersonate(accountingOracleAddress, ether("1000.0"));
-  //
-  //     await expect(lido.connect(accountingOracleSigner).handleOracleReport(...report())).not.to.emit(
-  //       postTokenRebaseReceiver,
-  //       "Mock__PostTokenRebaseHandled",
-  //     );
-  //   });
-  //
-  //   it("Reverts if there are withdrawal batches submitted and `checkSimulatedShareRate` fails", async () => {
-  //     await oracleReportSanityChecker.mock__checkSimulatedShareRateReverts(true);
-  //
-  //     await expect(
-  //       lido.handleOracleReport(
-  //         ...report({
-  //           withdrawalFinalizationBatches: [1n],
-  //         }),
-  //       ),
-  //     ).to.be.reverted;
-  //   });
-  //
-  //   it("Does not revert if there are no withdrawal batches submitted but `checkSimulatedShareRate` fails", async () => {
-  //     await oracleReportSanityChecker.mock__checkSimulatedShareRateReverts(true);
-  //
-  //     await expect(lido.handleOracleReport(...report())).not.to.be.reverted;
-  //   });
-  //
-  //   it("Returns post-rebase state", async () => {
-  //     const postRebaseState = await lido.handleOracleReport.staticCall(...report());
-  //
-  //     expect(postRebaseState).to.deep.equal([await lido.getTotalPooledEther(), await lido.getTotalShares(), 0n, 0n]);
-  //   });
-  // });
-});
+import { expect } from "chai";
+import { ZeroAddress } from "ethers";
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+
+import {
+  Accounting,
+  Burner__MockForAccounting,
+  Burner__MockForAccounting__factory,
+  IPostTokenRebaseReceiver,
+  Lido__MockForAccounting,
+  Lido__MockForAccounting__factory,
+  LidoExecutionLayerRewardsVault__MockForLidoAccounting,
+  LidoExecutionLayerRewardsVault__MockForLidoAccounting__factory,
+  LidoLocator,
+  OracleReportSanityChecker__MockForAccounting,
+  OracleReportSanityChecker__MockForAccounting__factory,
+  PostTokenRebaseReceiver__MockForAccounting__factory,
+  StakingRouter__MockForLidoAccounting,
+  StakingRouter__MockForLidoAccounting__factory,
+  WithdrawalQueue__MockForAccounting,
+  WithdrawalQueue__MockForAccounting__factory,
+  WithdrawalVault__MockForLidoAccounting,
+  WithdrawalVault__MockForLidoAccounting__factory,
+} from "typechain-types";
+import { ReportValuesStruct } from "typechain-types/contracts/0.8.9/oracle/AccountingOracle.sol/IReportReceiver";
+
+import { certainAddress, ether, impersonate } from "lib";
+
+import { deployLidoLocator, updateLidoLocatorImplementation } from "test/deploy";
+
+describe("Accounting.sol:report", () => {
+  let deployer: HardhatEthersSigner;
+
+  let accounting: Accounting;
+  let postTokenRebaseReceiver: IPostTokenRebaseReceiver;
+  let locator: LidoLocator;
+
+  let lido: Lido__MockForAccounting;
+  let elRewardsVault: LidoExecutionLayerRewardsVault__MockForLidoAccounting;
```

Comment:
We can remove excess dependencies if testing only Accounting: elRewardsVault and withdrawalVault